site/source/tutorials/005_form_widgets/index.md

```diff
@@ -1,26 +1,159 @@
+
+The method starts by decomposing the properties down in to local constants. We still need to define those properties, so update the `WorkerFormProperties` interface to include them:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:9-19 %}
+
+There are three types of properties that we are using in this form. The `firstName`, `lastName` and `email` properties are going to set the values that are displayed in the form fields. The `firstNameInvalid`, `lastNameInvalid` and `emailInvalid` properties will be used to indicate if a field contains invalid data (e.g. an invalid email address in the `email` field). Finally, the `onChange`, `onBlur` and `onSubmit` properties expose the events that the `WorkerForm` widget can emit. To see how these different property types are used, let's examine the properties that are being passed into first `TextInput` widget:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:51-69 %}
+
+The first thing that we see is a `key` property. As mentioned before, a key is necessary whenever the same type of widget or virtual DOM element will be rendered by a widget. After that comes the `invalid` property. Setting this property to `true` will update the widget's appearance to indicate that the widget contains invalid data. Next comes the the `label`, `placeholder`, and `type` fields which map to the expected properties.
+
+The `value` property renders the value of the field that is passed into the widget via its properties. Notice that there is no code that manipulates this value within the widget. As parts a [reactive framework](https://en.wikipedia.org/wiki/Reactive_programming), Dojo 2 widgets don't normally update their own state. Rather, they inform their parents that a change has occurred via events or some other mechanism. The parent will then pass updated properties back into the widget after all of the changes have been processed. This allows Dojo 2 applications to centralize data and keep the entire application synchronized.
+
+Finally, the `onChange` and `onBlur` properties are populated with [arrow functions](https://www.typescriptlang.org/docs/handbook/functions.html) that call the corresponding `WorkerForm` event handler. This is another common pattern within Dojo 2 applications - the `WorkerForm` doesn't expose any of the components that it is using to build the form. Rather, it manages its children internally and, if necessary, raises its events to inform its parents of any changes they might be interested in. This frees the consumers of the `WorkerForm` widget from having to understand the internal structure of the widget. Additionally, it allows the `WorkerForm` to change its implementation without affecting its parents as long as it continues to honor the `WorkerFormProperties` interface.
```

Comment:
"honor" -> perhaps follow instead of honor which feels like a potentially loaded term?